pkg/gpu/nvidia/metrics/metrics.go

```diff
@@ -26,6 +26,29 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
+type metricsCollector interface {
+	collectGPUDevice(deviceName string) (*nvml.Device, error)
+	collectStatus(*nvml.Device) (status *nvml.DeviceStatus, err error)
+	collectDutyCycle(string, time.Duration) (uint, error)
+}
+
+var g metricsCollector
```

Comment:
nit: maybe a longer name for this package level global variable?

Reply:
changed to gmc...not good with names...Do you have better suggestions?

Reply:
gmc looks good to me